### PR TITLE
[feature][doc] Add docs about how to use basic authentication

### LINKED ISSUE #14661: [Doc] Add doc for basic authentication
### What issue do you find in Pulsar docs?

When updating doc for security with @tuteng, we found that Pulsar supports basic authentication as one of the authentication providers. However, in the Security chapter, there is no topic talking about basic authentication like those for TLS, Athenz and etc.
image

### What is your suggestion?

Confirmed with @codelipenghui:
In parallel with other available authentication providers, add a new topic for basic authentication, introducing the required concept and configurations.

### Do you have any references?

N/A

### Would you like to fix this issue?

In the backlog.

### Note

- [X] I have researched my question.

### PR DESCRIPTION
**Pull request by momo-jun**
Fixes #14661 

### Modifications

1. Add a new topic about how to use basic authentication
2. Sync the topic in the left navigation and security overview

The preview looks good.
image
image
image
image

Note that the changes need to be applied to more historical versions since 2.8 as soon as it gets approved.

### Documentation
  
- [x] `doc`

### CODE REVIEW

**Comment by momo-jun** [comment]
Ping @nodece @codelipenghui @Anonymitaet for review. Thanks.

File: site2/docs/security-basic-auth.md [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+   brokerClientAuthenticationParameters={"userId":"superuser","password":"admin"}
+
+   # Whether client authorization credentials are forwarded to the broker for re-authorization.
+   # Authentication must be enabled via authenticationEnabled=true for this to take effect.
+   forwardAuthorizationCredentials=true
+   ```
+
+2. Set an environment variable named `pulsar.auth.basic.conf` and the value is `.htpasswd`. Pulsar reads this environment variable to implement HTTP basic authentication.
+
+## Configure basic authentication through CLI tools
```
Comment (nodece) [inline site2/docs/security-basic-auth.md:100]: through -> in?

File: site2/docs/security-basic-auth.md [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+   ```
+
+## Create your authentication file
+
+:::note
+Currently, you can use MD5 (recommended) and CRYPT encryption to authenticate your password.
+:::
+
+Create a password file named `.htpasswd` with a user account `superuser/admin`:
+* Use MD5 encryption (recommended):
```
Comment (Anonymitaet) [inline site2/docs/security-basic-auth.md:37]: This part is not shown in the preview screenshot. Does it render correctly (there is no empty line between two sentences)?

Comment (Anonymitaet) [reply site2/docs/security-basic-auth.md:37]: Can you attach the preview screenshot for all changes?

Comment (momo-jun) [reply site2/docs/security-basic-auth.md:37]: I've attached all screenshots.

**Review by nodece — APPROVED** [review]

File: site2/docs/security-basic-auth.md (hunk 1 above)
Comment (nodece) [reply site2/docs/security-basic-auth.md:100]: This part still miss.

Comment (momo-jun) [reply site2/docs/security-basic-auth.md]: fixed

**Pull request merged**